roles/mongodb/tasks/mongodb_yum.yaml

```diff
@@ -1,30 +1,41 @@
 ---
-- include: mongodb_rhel_dependencies.yaml
+- name: yum | Install mongodb dependencies
+  become: yes
+  yum: name={{item}} state=present
+  with_items:
+    # Failed to validate the SSL certificate for www.mongodb.org:443. Make sure your managed systems have a valid CA certificate installed. If the website serving the url uses SNI you need python >= 2.7.9 on your managed machine or you can install the `urllib3`, `pyopenssl`, `ndg-httpsclient`, and `pyasn1` python modules to perform SNI verification in python >= 2.6. You can use validate_certs=False if you do not need to confirm the servers identity but this is unsafe and not recommended. Paths checked for this platform: /etc/ssl/certs, /etc/pki/ca-trust/extracted/pem, /etc/pki/tls/certs, /usr/share/ca-certificates/cacert.org, /etc/ansible
+    - python-urllib3
+    - pyOpenSSL
+    - python-pyasn1
+    - python-ndg_httpsclient
 
 - name: yum | Add mongodb key
   become: yes
   rpm_key:
-    key: https://www.mongodb.org/static/pgp/server-3.4.asc
+    key: https://www.mongodb.org/static/pgp/server-{{ mongodb_major_minor_version }}.asc
     state: present
   tags: [databases, mongodb]
 
 - name: yum | Add mongodb repository
   become: yes
   yum_repository:
-    name: mongodb-org-3.4
+    name: mongodb-org-{{ mongodb_major_minor_version }}
     description: MongoDB Repository
     gpgcheck: yes
     enabled: yes
-    baseurl: https://repo.mongodb.org/yum/redhat/$releasever/mongodb-org/3.4/x86_64/
-    gpgkey: https://www.mongodb.org/static/pgp/server-3.4.asc
+    baseurl: https://repo.mongodb.org/yum/redhat/$releasever/mongodb-org/{{ mongodb_major_minor_version }}/x86_64/
+    gpgkey: https://www.mongodb.org/static/pgp/server-{{ mongodb_major_minor_version }}.asc
     state: present
   tags: [databases, mongodb]
 
 - name: yum | Install mongodb
   become: yes
   yum:
-    name: mongodb-org
+    name: mongodb-org-{{ mongodb_version }}*
```

Comment:
Shouldn't this be same as: https://github.com/StackStorm/ansible-st2/pull/160/files#diff-7bdf986c5bd89f75152b5fcc42b77680R24 , explicitly install dependencies.

Reply:
No, I checked under CentOS and it pulls correct package version dependencies.

So there is a diff in logic for installing `deb` and `rpm` mongo packages.